### DISCUSSION TITLE
Multiple db connections, make:entity not working and strange doctrine annotations

### DISCUSSION THREAD

**Discussion by FrankBusch68** [symfony/symfony#49329] [Q&A] [answered]
Hi community,
I have to create a project where I have to handle different databases and therefore different connections.

I followed [https://symfony.com/doc/current/doctrine/multiple_entity_managers.html](https://symfony.com/doc/current/doctrine/multiple_entity_managers.html)

My config/packages/doctrine.yaml looks like this:
```
doctrine:
  dbal:
    default_connection: default
    connections:
      default:
        # configure these for your database server
        url: '%env(resolve:DATABASE_URL)%'
        driver: 'pdo_mysql'
        server_version: '5.7'
        charset: utf8mb4
      messaging:
        # configure these for your database server
        url: '%env(resolve:DATABASE_MESSAGING_URL)%'
        driver: 'pdo_mysql'
        server_version: '5.7'
        charset: utf8mb4
  orm:
    auto_generate_proxy_classes: true
    enable_lazy_ghost_objects: true
    entity_managers:
      default:
        connection: default
        mappings:
          Main:
            is_bundle: false
            type: annotation
            dir: '%kernel.project_dir%/src/Entity/Main'
            prefix: 'App\Entity\Main'
            alias: Main
        naming_strategy: doctrine.orm.naming_strategy.underscore_number_aware
      messaging:
        connection: messaging
        mappings:
          Messaging:
            is_bundle: false
            type: annotation
            dir: '%kernel.project_dir%/src/Entity/Messaging'
            prefix: 'App\Entity\Messaging'
            alias: Messaging
        naming_strategy: doctrine.orm.naming_strategy.underscore_number_aware
        auto_mapping: true
when@test:
  doctrine:
    dbal:
      # "TEST_TOKEN" is typically set by ParaTest
      dbname_suffix: '_test%env(default::TEST_TOKEN)%'

when@prod:
  doctrine:
    orm:
      auto_generate_proxy_classes: false
      proxy_dir: '%kernel.build_dir%/doctrine/orm/Proxies'
      query_cache_driver:
        type: pool
        pool: doctrine.system_cache_pool
      result_cache_driver:
        type: pool
        pool: doctrine.result_cache_pool

  framework:
    cache:
      pools:
        doctrine.result_cache_pool:
          adapter: cache.app
        doctrine.system_cache_pool:
          adapter: cache.system

```
This seems to be syntactically ok... At least there is no error when starting a console statement.

Now I do a
`    symfony console make:entity Messaging\\Message`
and get
```
created: src/Entity/Messaging/Message.php
 created: src/Repository/Messaging/MessageRepository.php

 [ERROR] Only attribute mapping is supported by make:entity, but the <info>App\Entity\Messaging\Message</info> class uses a
         different format. If you would like this command to generate the properties & getter/setter methods, add your
         mapping configuration, and then re-run this command with the <info>--regenerate</info> flag.

```
This file was generated at src/Entity/Messaging... 
```
<?php

namespace App\Entity\Messaging;

use App\Repository\Messaging\MessageRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MessageRepository::class)]
class Message
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    public function getId(): ?int
    {
        return $this->id;
    }
}

```
So I tried it again to at least add some more properties and got
```
In DoctrineHelper.php line 91:

  Cannot find the entity manager for class "App\Entity\Messaging\Message"

```
So I tried
`     symfony console make:entity App\\Entity\\Messaging --regenerate` 
as mentioned in the first error message and got
`     [ERROR] No entities were found in the "App\Entity\Messaging" namespace.`
even though this is the namespace in the generated file.

After all that... these are my questions:

1. What did I do wrong?
2. What did I forget to config?
3. What do I have to change?
4. Why are there no "nice" PHPDoc-annotations? (Neither with doctrine/annotations installed, nore without)

And by the way: Sorry for bad formatting... I'm very tired after all the tries I did ;)

**Comment by ahundiak** [comment, marked as answer]
It is a bit hard to explain but basically we are in the midst of transiting from annotations to attributes.  make:entity really only wants to generate new entities with attributes but sort of supports editing existing annotation based entities.  So messy things happen.

make:entity tries to use the fully qualified entity class name to determine which entity manager will manage it and then pull the type attribute to determine if annotations or attributes should be used.  You can create a new 6.2 project and then just edit doctrine.yaml and set type to attribute.  At which point make:entity will fail with with the message you are getting:
```
[ERROR] Only attribute mapping is supported by make:entity, but the <info>App\Entity\Blue</info> class uses a different
         format. If you would like this command to generate the properties & getter/setter methods, add your mapping    
         configuration, and then re-run this command with the <info>--regenerate</info> flag.  
```
So no new annotated entities.  If you manually create an entity file with annotated properties then `make:entity --regenerate` will add the getter/setters methods.  Which I suppose helps in some cases.

Bottom line: if you want to use make:entity then switch to attributes.

**Reply by FrankBusch68** [reply]
Damn, setting the type to attributes really did the trick. Generating, regenerating, selecting.. it all works. 
Maybe you should forward this to sensio in order to change the example on [https://symfony.com/doc/current/doctrine/multiple_entity_managers.html](https://symfony.com/doc/current/doctrine/multiple_entity_managers.html).
Thanks for help
